### PR TITLE
Pledge PDF generation (rebased)

### LINKED ISSUE #345: Generate and download PDF of pledge
@lwaldsc @NerdyGirl537 can you all post a screenshot of a pledge form with fake information in it, so we can incorporate that into the pledge modal?

### PR DESCRIPTION
**Pull request by colinxfleming**
I rule and have completed some work on Case Manager that's ready for review!

PDF work continues apace. See #956 , #974 , #917 .

This pull request makes the following changes:
* Last set of changes! Hopefully!
* Has the bulk of styling, testing, and workflow in order.

I got approval from DCAF CMDs and the accountant to roll this out as is for the time being, though we may need to add some language for overruled pledges once they can agree on what the proper workflow is on the business side.

It relates to the following issue #s: 
* Fixes #345

### CODE REVIEW

File: app/services/pledge_form_generator.rb [hunk 1]
```diff
@@ -0,0 +1,144 @@
+class PledgeFormGenerator
+  attr_reader :patient, :user, :case_manager_name
+
+  def initialize(user, patient, case_manager_name)
+    @patient = patient
+    @user = user
+    @case_manager_name = case_manager_name.to_s
```
Comment (colinxfleming) [inline app/services/pledge_form_generator.rb]: @feminismisawesome astutely noted that this should be `to_s`'d in the controller in https://github.com/colinxfleming/dcaf_case_management/pull/956/files#r108524143 .

File: test/controllers/patients_controller_test.rb [hunk 2]
```diff
@@ -80,7 +80,7 @@ class PatientsControllerTest < ActionController::TestCase
       @date = 5.days.from_now.to_date
       @payload = {
         appointment_date: @date.strftime('%Y-%m-%d'), name: 'Susie Everyteen 2',
-        pregnancy: { resolved_without_dcaf: true }
+        pregnancy: {resolved_without_dcaf: true}
```
Comment (colinxfleming) [inline test/controllers/patients_controller_test.rb]: revert this

**Comment by colinxfleming** [comment]
I **think** the last step here is to plug this into #975 ?

**Comment by colinxfleming** [comment]
I think this is pretty much ready to ship, at long last! @lomky @FeminismIsAwesome can I get a review on the code here?

File: app/services/pledge_form_generator.rb [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    @patient = patient
+    @user = user
+    @case_manager_name = case_manager_name
+  end
+
+  def patient_amount
+    number_to_currency(@patient.pregnancy.dcaf_soft_pledge, precision: 0)
+  end
+
+  # TODO when clinics actually exist instead of just name, do this
```
Comment (FeminismIsAwesome) [inline app/services/pledge_form_generator.rb]: should probably remove the TODO since it works now :P

**Comment by FeminismIsAwesome** [comment]
@colinxfleming had one minor commentary about a TODO, but otherwise lookin' good! go teamwork! 👍

File: app/services/pledge_form_generator.rb (hunk 3 above)
Comment (colinxfleming) [inline app/services/pledge_form_generator.rb]: haha right you are!

**Comment by colinxfleming** [comment]
awesome! Making that small fix, getting the build to pass, and then MERGING THIS UP

**Pull request merged**